calcom/cal.com
# feat: Implement flagsmith feature flags

## What does this PR do?

This PR introduces way to release features in a phased roll-out manner. Its a fully opt in feature such that if the env variable is set it fetches flags from flagsmith otherwise it fetches flags from our local db.

Fixes #12598

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update (probably)

## How should this be tested?

- Go to https://github.com/Flagsmith/flagsmith
-  you can either run their docker container or visit their hosted service
- create a new project, this will automatically generate a new api-key
- replace `FLAGSMITH_ENVIRONMENT_ID` in .env with that api-key
- create these flags in the dashboard, they should be exactly similar to the ones in our `Feature` table since it will replace those flags if the .env variable is set
![Screenshot 2024-01-26 at 5 28 49 PM](https://github.com/calcom/cal.com/assets/32706411/4f9edf9d-9e4f-4021-a5a3-6d8c63a2131d)
